kotlinKoans/src/main/java/com/epam/kotlinkoans/ii_collections/n22Fold.kt

```diff
@@ -15,6 +15,6 @@ fun whatFoldDoes(): Int {
 fun Shop.getSetOfProductsOrderedByEachCustomer(): Set<Product> {
     // Return the set of products that were ordered by each of the customers
     return customers.fold(allOrderedProducts, { orderedByAll, customer ->
-        todoCollectionTask()
-    })
+        orderedByAll.intersect( customer.orders.flatMap { it.products }.toSet() ) })
```

Comment:
`customer.orders.flatMap { it.products }.toSet()` could be replaced with `customer.orderedProducts`